### ISSUE TITLE
SSHFS Seg Faults on Lollipop for Moto X 2013 (ghost)

### ISSUE THREAD

**Issue by Drakonas** [greyltc/android_external_sshfs#5]
SSHFS properly mounts the folder, but if you try to open anything but a folder, the SSH server successfully opens and closes the file, but the SSHFS client reads a few bits of the file, and then immediately ends in a memory violation sig 11 seg fault.

Command used to connect:
`sshfs drakonas@192.168.1.16:/ /system/sshfs -o ro -o allow_other -o follow_symlinks -o StrictHostKeyChecking=no -o reconnect -o TCPKeepAlive=no`

[Full catlog from before issue occurs to after issue occurs](http://db.3rdn3rd.com/txt/catlog/2015-06-09-06-05-35.txt)
[Clipped catlog from where problem started and ended](http://db.3rdn3rd.com/txt/catlog/2015-06-09-06-15-03.txt)
[Filtered catlog by 'ssh'](http://db.3rdn3rd.com/txt/catlog/2015-06-09-06-15-25.txt)

Note: I might have been using Enforcing SELinux mode at the time of the catlog saves, but the same issue happens in Permissive SELinux mode.

2nd Note: See my first issue that explains related problems, and more details on my setup: https://github.com/l3iggs/android_external_sshfs/issues/4

**Comment by l3iggs** [comment]
I hope that 2829f966843041ec1c4454aab6ed5a7874115f53 has fixed this. Please let me know if it has.

**Comment by Drakonas** [comment]
I am getting the same problem even with your new commit. I double checked to make sure I have your new changes pulled. Still seems to crash when I try to read a file.

Perhaps I should attempt updating my rom. While the rom I am using is the best daily driver rom I've used since 4.4.2 stock-based rom, the version I am using is based on 5.1.1r3. The rom has been updated to 5.5.2 since then. I am not sure if my rom might be producing other inconsistencies. I would prefer not to update because this is the most stable my phone's been since a year ago, but I may have to to make sure the rom isn't at fault.
